src/main/java/com/yelp/nrtsearch/server/luceneserver/search/collectors/MyTopSuggestDocsCollector.java

```diff
@@ -79,7 +93,119 @@ public TopSuggestDocs reduce(Collection<TopSuggestDocsCollector> collectors)
       for (TopSuggestDocsCollector collector : collectors) {
         topDocs[i++] = collector.get();
       }
-      return TopSuggestDocs.merge(numHitsToCollect, topDocs);
+
+      // create a map with docId for key and the best SuggestScoreDoc with the docId
+      // the "best" suggestScoreDoc is the one with the highest score
+      Map<Integer, TopSuggestDocs.SuggestScoreDoc> topSuggestScoreDocsUniqueByDoc = new HashMap<>();
+      for (TopSuggestDocs topSuggestDocs : topDocs) {
+        for (TopSuggestDocs.SuggestScoreDoc currentSearchSuggestDoc :
+            topSuggestDocs.scoreLookupDocs()) {
+          int docId = currentSearchSuggestDoc.doc;
+          if (topSuggestScoreDocsUniqueByDoc.containsKey(docId)) {
+            TopSuggestDocs.SuggestScoreDoc existingSearchSuggestDocForDocId =
+                topSuggestScoreDocsUniqueByDoc.get(docId);
+            if (existingSearchSuggestDocForDocId.score < currentSearchSuggestDoc.score) {
+              topSuggestScoreDocsUniqueByDoc.put(docId, currentSearchSuggestDoc);
+            }
+          } else {
+            topSuggestScoreDocsUniqueByDoc.put(docId, currentSearchSuggestDoc);
+          }
+        }
+      }
+
+      // retrieve the top this.numHitsToCollect suggestScoreDocs, sorting them by score
+      TopSuggestDocs.SuggestScoreDoc[] suggestScoreDocs =
+          topSuggestScoreDocsUniqueByDoc.values().stream()
+              .sorted((o1, o2) -> -1 * Float.compare(o1.score, o2.score)) // highest score first
+              .limit(this.numHitsToCollect)
+              .toArray(TopSuggestDocs.SuggestScoreDoc[]::new);
```

Comment:
I don't think all of this is needed if deduplication is done by id elsewhere. A given doc id will only exist in a single segment, so there will be no duplication between collectors.